tests/src/Unit/CreateMembershipTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Drupal\Tests\og\Unit;
+
+
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\og\Og;
+use Drupal\og\GroupManager;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Tests create membership helper function.
+ *
+ * @group og
+ * @coversDefaultClass \Drupal\og\Og
+ */
+class CreateMembershipTest extends UnitTestCase {
+
+  /**
+   * The entity manager used for testing.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface|\Prophecy\Prophecy\ObjectProphecy
+   */
+  protected $entityManager;
+
+  /**
+   * The entity storage prophecy used in the test.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface|\Prophecy\Prophecy\ObjectProphecy
+   */
+  protected $entityStorage;
+
+  /**
+   * The mocked group manager.
+   *
+   * @var \Drupal\og\GroupManager|\Prophecy\Prophecy\ObjectProphecy
+   */
+  protected $groupManager;
+
+  /**
+   * A mocked test user.
+   *
+   * @var \Drupal\Core\Session\AccountInterface|\Prophecy\Prophecy\ObjectProphecy
+   */
+  protected $user;
+
+  /**
+   * The entity type ID of the test group.
+   *
+   * @var string
+   */
+  protected $entityTypeId;
+
+  /**
+   * The bundle ID of the test group.
+   *
+   * @var string
+   */
+  protected $bundle;
+
+  /**
+   * The mocked test group.
+   *
+   * @var \Drupal\Core\Entity\EntityInterface|\Prophecy\Prophecy\ObjectProphecy
+   */
+  protected $group;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    $this->entityTypeId = $this->randomMachineName();
+    $this->bundle = $this->randomMachineName();
+
+    $this->groupManager = $this->prophesize(GroupManager::class);
+    $this->groupManager->isGroup($this->entityTypeId, $this->bundle)->willReturn(TRUE);
+
+    $this->entityStorage = $this->prophesize(EntityStorageInterface::class);
+
+    $this->entityManager= $this->prophesize(EntityManagerInterface::class);
+    $this->entityManager->getStorage($this->entityTypeId)
+      ->willReturn($this->entityStorage->reveal());
+
+    $this->entityManager->getEntityTypeFromClass('Drupal\og\Entity\OgMembership')
+      ->willReturn($this->entityTypeId);
+
+    $entity = $this->prophesize(EntityInterface::class);
+    $this->entityStorage->create()
```

Comment:
I must be doing something wrong here, as I get:

![d8_dev_ _amitai_amitais-macbook-pro_ _zsh_ _112x34](https://cloud.githubusercontent.com/assets/125707/17113516/ca54195c-52b3-11e6-9b9e-127112568a4e.jpg)

I wonder if indeed should do it with unit tests, or just go with Kernel.